src/test/java/io/vertx/core/BlockedThreadCheckerTest.java

```diff
@@ -119,4 +122,36 @@ public void start() throws InterruptedException {
     await();
     blockedThreadWarning.expectMessage("vert.x-worker-thread", maxWorkerExecuteTime, maxWorkerExecuteTimeUnit);
   }
+
+  @Test
+  public void testCustomThreadBlockedHandler() throws Exception {
+    CountDownLatch cl = new CountDownLatch(1);
+    Verticle verticle = new AbstractVerticle() {
+      @Override
+      public void start() throws InterruptedException {
+        Thread.sleep(3000);
+        testComplete();
+      }
+    };
+    long maxWorkerExecuteTime = 1000;
+    long warningExceptionTime = 2;
+    TimeUnit maxWorkerExecuteTimeUnit = MILLISECONDS;
+    TimeUnit warningExceptionTimeUnit = SECONDS;
+    VertxOptions vertxOptions = new VertxOptions();
+    vertxOptions.setMaxWorkerExecuteTime(maxWorkerExecuteTime);
+    vertxOptions.setMaxWorkerExecuteTimeUnit(maxWorkerExecuteTimeUnit);
+    vertxOptions.setWarningExceptionTime(warningExceptionTime);
+    vertxOptions.setWarningExceptionTimeUnit(warningExceptionTimeUnit);
+    Vertx newVertx = vertx(vertxOptions);
+    ((VertxInternal) newVertx).blockedThreadChecker().setThreadBlockedHandler(bte -> {
+      Assert.assertEquals(NANOSECONDS.convert(maxWorkerExecuteTime, maxWorkerExecuteTimeUnit), bte.maxExecTime);
+      Assert.assertEquals(NANOSECONDS.convert(warningExceptionTime, warningExceptionTimeUnit), bte.warningExceptionTime);
+      cl.countDown();
+    });
+    DeploymentOptions deploymentOptions = new DeploymentOptions();
+    deploymentOptions.setWorker(true);
+    newVertx.deployVerticle(verticle, deploymentOptions);
+    await();
+    awaitLatch(cl);
```

Comment:
avoid using a latch here, instead use `waitFor(2)` at the beginning of the test and `complete()` twice (one of `testComplete()` and one to replace the latch count down.

Reply:
@vietj I tried to do it this way initially, but the `testComplete()` and `assert*` methods provided by the base class have checks to make sure they are called from an event loop thread, which means the test fails when they get called from the BlockedThreadChecker thread.

I could use `runOnContext` to get back on an event loop thread inside of the `threadBlockedHandler`, though. Do you prefer that to using the `CountdownLatch`?